airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py

```diff
@@ -228,7 +264,17 @@ def read(self) -> Iterable[Record]:
         Otherwise, the message will be emitted on the message repository.
         """
         try:
-            for record_data in self._stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=copy.deepcopy(self._slice)):
+            # using `stream_state=self._state` have a very different behavior than the current one as today the state is updated slice
```

Comment:
is the concern that we don't control how the state is accessed so it's possible for a stream's behavior to change now that different slices can be updated concurrently?

Reply:
The concern is that today, `stream_instance.get_updated_state` is called on every record and `read_records ` on every slice. The implication is that the argument `stream_state` passed to `read_records` will have the value after the last `stream_instance.get_updated_state` of the previous slice. For Concurrent CDK, this is not possible as slices are processed in an unordered way. Hence, this is a change of behavior that could have impacts is the stream uses `stream_state` for it's content (Stripe only does `if not stream_state` so whatever the content as long as there is a state, we're fine)